core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -99,4 +102,36 @@ class BpmBillingProjectCreator(billingRepository: BillingRepository,
     } catch {
       case exception: Exception => Future.failed(exception)
     }
+
+  override def preDeletionSteps(projectName: RawlsBillingProjectName, ctx: RawlsRequestContext): Future[Unit] =
+    for {
+      _ <- billingRepository.getCreationStatus(projectName).flatMap {
+        case CreationStatuses.CreatingLandingZone =>
+          Future.failed(
+            new BillingProjectDeletionException(
+              RawlsErrorReport(
+                s"Billing project ${projectName.value} cannot be deleted because its landing zone is still being created"
+              )
+            )
+          )
+        case _ => Future.successful()
+      }
+      _ <- billingRepository.getLandingZoneId(projectName).flatMap {
+        case Some(landingZoneId) =>
+          val landingZoneResponse = blocking {
+            workspaceManagerDAO.deleteLandingZone(UUID.fromString(landingZoneId), ctx)
```

Comment:
@aherbst-broad delete is async…. it will just start the job, so should be fast. We decided in the ticket that we are going "fire and forget", and not worry if the deletion ends up failing for some reason.

Reply:
I will add a code comment that the operation is asynchronous, and we aren't tracking whether it ultimately succeeds.

Reply:
Fixed